### PR TITLE
Added documentation on how to provide access token

### LINKED ISSUE #15466: Integrate Custom Views with Ledger API authorization

### PR DESCRIPTION
**Pull request by chunlokling-da**
Fixes digital-asset/daml#15466

### CODE REVIEW

File: REFERENCE.md [hunk 1]
```diff
@@ -253,4 +253,37 @@ projection {
 }
 ```
 
-You can override this dispatcher through the application.conf in your application.
\ No newline at end of file
+You can override this dispatcher through the application.conf in your application.
+
+## Ledger API Authorization
+
+Client of ledger provides an access token when authorization is required by the Ledger.
```
Comment (S11001001) [inline REFERENCE.md]:
```suggestion
The client must provide an access token when authorization is required by the Ledger.
```

File: REFERENCE.md [hunk 2]
```diff
@@ -253,4 +253,37 @@ projection {
...
+Detail of Ledger Authorization please refer to [Ledger Authorization documentation](https://docs.daml.com/app-dev/authorization.html)
```
Comment (S11001001) [inline REFERENCE.md]:
```suggestion
For details of ledger authorization, please refer to [Ledger Authorization documentation](https://docs.daml.com/app-dev/authorization.html).
```

File: REFERENCE.md [hunk 3]
```diff
@@ -253,4 +253,37 @@ projection {
...
+
+### Provide access token to custom-view library
+
+Application can provide an access token when setting up the clientSetting.
```
Comment (S11001001) [inline REFERENCE.md]:
```suggestion
Applications can provide an access token when setting up the client. 
```

File: REFERENCE.md [hunk 4]
```diff
@@ -253,4 +253,37 @@ projection {
...
+val clientSettings = GrpcClientSettings
+  .connectToServiceAt(host, port.value)
+  .withCallCredentials(new LedgerCallCredentials(accessToken))
+val source = BatchSource.events(clientSettings)
+val control = Projection.project(source, exercisedEvents)(f)
+```
+
+### Provide a newly retrieved access token when the existing one expired
+
+When the access token is expired, application can retrieve a new access token with the stored refresh token.
```
Comment (S11001001) [inline REFERENCE.md]:
```suggestion
When an access token is expired, an application can retrieve a new access token with the stored refresh token.
```

File: REFERENCE.md [hunk 5]
```diff
@@ -253,4 +253,37 @@ projection {
...
+Detail of refresh access token please refer to [Ledger auth-middleware documentation](https://docs.daml.com/tools/auth-middleware/index.html#refresh-access-token)
```
Comment (S11001001) [inline REFERENCE.md]:
```suggestion
For details on the refresh token, please refer to [Ledger auth-middleware documentation](https://docs.daml.com/tools/auth-middleware/index.html#refresh-access-token).
```

File: REFERENCE.md [hunk 6]
```diff
@@ -253,4 +253,37 @@ projection {
...
+Application can cancel the running projection and re-create a new one with the new access token.
```
Comment (S11001001) [inline REFERENCE.md]:
```suggestion
With the new access token, an application can cancel the running projection and re-create a new one using the new token.
```

**Review by S11001001 — APPROVED** [review]
@digital-asset/daml-docs may also have some input.

**Comment by ray-roestenburg-da** [comment]
@chunlokling-da docs should be added to daml/docs. I've probably confused everyone by leaving this reference.md around in the repo.

**Comment by chunlokling-da** [comment]
@ray-roestenburg-da Sure, I will make another PR to add the doc to daml docs instead

**Comment by chunlokling-da** [comment]
add the doc to daml docs instead

**Comment by chunlokling-da** [comment]
created https://github.com/digital-asset/daml/pull/15783

**Pull request closed without merging**